Modules/_winapi.c

```diff
@@ -1043,9 +1046,27 @@ _winapi_CreateProcess_impl(PyObject *module, Py_UNICODE *application_name,
 
     si.lpAttributeList = attribute_list.attribute_list;
 
+    // copy command line since CreateProcessW can change the content of the buffer
```

Comment:
I wouldn't worry about special casing long strings. I'd simply change the clinic argument to `unicode`, so we instead have `PyObject *command_line`. Then create the copy via `command_line_copy = PyUnicode_AsWideCharString(command_line, NULL)`.